src/rdb.c

```diff
@@ -3100,9 +3144,22 @@ int rdbLoadRioWithLoadingCtx(rio *rdb, int rdbflags, rdbSaveInfo *rsi, rdbLoadin
             } else if (!strcasecmp(auxkey->ptr, "redis-bits")) {
                 /* Just ignored. */
             } else {
-                /* We ignore fields we don't understand, as by AUX field
-                 * contract. */
-                serverLog(LL_DEBUG, "Unrecognized RDB AUX field: '%s'", (char *)auxkey->ptr);
+                /* Check if this is a dynamic aux field */
+                int handled = 0;
+                if (rdbAuxFields != NULL) {
+                    dictEntry *de = dictFind(rdbAuxFields, auxkey->ptr);
+                    if (de != NULL) {
+                        handled = 1;
+                        rdbAuxFieldCodec *codec = (rdbAuxFieldCodec *)dictGetVal(de);
+                        if (codec->decoder(rdbflags, auxval->ptr) < 0) goto eoferr;
```

Comment:
auxval is an robj, which is always backed by an sds in `->ptr`. So I think the decoder implementation can be an SDS. In your case I don't think it matters, but in the general case someone might want to know the length.